### PR TITLE
devtools: Auto-set branch to merge to in github-merge

### PR DESCRIPTION
**Pull request by laanwj**
As we are already using the API to retrieve the pull request title, also retrieve the base branch.

This makes sure that pull requests for 0.12 automatically end up in 0.12, and pull requests for master automatically end up in master, and so on.

It prints the branch it is about to merge into. It is still possible to override the branch from the command line or using the `githubmerge.branch` git option.

### CODE REVIEW

**Comment by sipa** [comment]
Concept ACK; this has been a feature I've wanted for a while.

**Comment by jonasschnelli** [comment]
Nice! Concept ACK.
Great safety measure also.

**Comment by MarcoFalke** [comment]
utACK c1cff21

File: contrib/devtools/github-merge.py [hunk 1]
```diff
@@ -105,9 +105,23 @@ def main():
...
+    title = info['title']
+    # precedence order for destination branch argument:
+    #   - command line argument
+    #   - githubmerge.branch setting
+    #   - base branch for pull (as retrieved from github)
+    #   - 'master'
     branch = args.branch
+    branch = opt_branch if branch is None else branch
+    branch = info['base']['ref'] if branch is None else branch
+    branch = 'master' if branch is None else branch
```
Comment (MarcoFalke) [inline contrib/devtools/github-merge.py]: Nit: Couldn't this be just one line?

Something like

```py
branch = args.branch or opt_branch or info['base']['ref'] or 'master'

Comment (laanwj) [inline contrib/devtools/github-merge.py]: That's much shorter, also seems to work:
```
>>> None or 'a' or 'b' or 'c'
'a'
>>> None or None or 'b' or 'c'
'b'
>>> None or None or None or 'c'
'c'
>>> 'f' or None or None or 'c'
'f'
```
I think the subtle difference in this case is that '' is seen as False (hence the use of `is None`. But doesn't matter here, an empty branch name is pointless.

**Comment by btcdrak** [comment]
utACK 10d3ae1

**Comment by laanwj** [comment]
I've been using this version for a few days, both on master and 0.12, seems to work.
Tested ACK 10d3ae1

**Pull request merged**